.pipelines/mac-ios-spm-extensions-validation-pipeline.yml

```diff
@@ -0,0 +1,101 @@
+  jobs:
+  - job: j
+    displayName: "Test with ORT Extensions native pod"
+
+    pool:
+      vmImage: "macOS-13"
+
+    variables:
+      xcodeVersion: "14.3"
+      artifactsName: "ios_packaging_artifacts_full"
+      artifactsNameForExt: "ios_packaging_artifacts"
+
+    timeoutInMinutes: 60
+
+    steps:
+    - template: templates/use-xcode-version.yml
+      parameters:
+        xcodeVersion: ${{ variables.xcodeVersion }}
+
+    - script: |
+        mkdir tmp
+        cd tmp
+        git clone -n --depth=1 --filter=tree:0 https://github.com/microsoft/onnxruntime.git
+        cd onnxruntime
+        git sparse-checkout set --no-cone objectivec
+        git checkout
+      workingDirectory: "$(Build.SourcesDirectory)"
+      displayName: "Sparse checkout objectivec/ folders from latest ORT main repository"
+    
+    - script: |
+        ls -R  "$(Build.SourcesDirectory)/tmp/onnxruntime"
+      workingDirectory: "$(Build.SourcesDirectory)/tmp"
+      displayName: "List sparse checkout repo contents"
+    
+    # Download artifacts for ORT C pod from iOS packaging pipeline
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        buildType: 'specific'
+        project: 'Lotus'
+        definition: 995          #'definitionid' is obtained from `System.DefinitionId` of ORT CI: onnxruntime-ios-packaging-pipeline
+        buildVersionToDownload: 'latest'
+        branchName: 'main'
+        targetPath: '$(Build.ArtifactStagingDirectory)'
+
+    - script: |
+        set -e -x
+        ls
+      workingDirectory: '$(Build.ArtifactStagingDirectory)/$(artifactsName)'
+      displayName: "List staged artifacts for ORT C Pod"
+
+    # Download artifacts for ORT Ext C pod from extensions iOS packaging pipeline
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        buildType: 'specific'
+        project: 'Lotus'
+        definition: 1206          #'definitionid' is obtained from `System.DefinitionId` of extensions CI: extensions.ios_packaging
+        buildVersionToDownload: 'specific'
+        branchName: 'yguo/change-header-path'   # TODO: update branch name here
+        pipelineID: 331190
+        targetPath: '$(Build.ArtifactStagingDirectory)'
+
+    - script: |
+        set -e -x
+        ls
+      workingDirectory: '$(Build.ArtifactStagingDirectory)/$(artifactsNameForExt)'
+      displayName: "List staged artifacts for ORT Ext C Pod"
+    
+    # Note: Running xcodebuild test on `onnxruntime-Package` scheme will perform swift unit tests for both OnnxRuntimeBindings
+    # and OnnxRuntimeExtensions targets.
+    - script: |
+        set -e -x
+        cd "$(Build.ArtifactStagingDirectory)/$(artifactsName)"
+        POD_ARCHIVE=$(find . -name "pod-archive-onnxruntime-c-*.zip")
+
+        shasum -a 256 "$(Build.ArtifactStagingDirectory)/$(artifactsName)/${POD_ARCHIVE}"
```

Comment:
Maybe echo a message saying why we checksum the file. 

Previously I added it to the CI output so it was easy to update Package.swift with the info for the actual release zip+checksum as it meant you didn't need to download the release and do this manually.